addons/account/models/account.py

```diff
@@ -226,20 +225,53 @@ def load(self, fields, data):
         return rslt
 
     @api.multi
+    def _toggle_reconcile_to_true(self):
+        query = """
+            UPDATE account_move_line
+                SET reconciled = CASE
+                    WHEN debit = 0 AND credit = 0 AND amount_currency = 0
+                    THEN true
+                    ELSE false
+                    END,
+                full_reconcile_id = NULL,
+                amount_residual = (debit-credit),
+                amount_residual_currency = amount_currency
+            WHERE account_id IN %s
+        """
+        self.env.cr.execute(query, [tuple(self.ids)])
+
+    @api.multi
+    def _toggle_reconcile_to_false(self):
+        partial_lines_count = self.env['account.move.line'].search_count([
+            ('account_id', 'in', self.ids),
+            ('full_reconcile_id', '=', False),
+            ('|'),
+            ('matched_debit_ids', '!=', False),
+            ('matched_credit_ids', '!=', False),
+        ])
+        if partial_lines_count > 0:
+            raise UserError(_('You cannot switch an account to not allow the reconciliation'
+                              'if some partial reconciliations are still pending.'))
+        query = """
+            UPDATE account_move_line
+                SET amount_residual = 0, amount_residual_currency = 0
+            WHERE full_reconcile_id = NULL AND account_id IN %s
+        """
+        self.env.cr.execute(query, [tuple(self.ids)])
+
+    @api.multi
     def write(self, vals):
         # Dont allow changing the company_id when account_move_line already exist
         if vals.get('company_id', False):
             move_lines = self.env['account.move.line'].search([('account_id', 'in', self.ids)], limit=1)
             for account in self:
                 if (account.company_id.id != vals['company_id']) and move_lines:
                     raise UserError(_('You cannot change the owner company of an account that already contains journal items.'))
-        # If user change the reconcile flag, all aml should be recomputed for that account and this is very costly.
-        # So to prevent some bugs we add a constraint saying that you cannot change the reconcile field if there is any aml existing
-        # for that account.
-        if vals.get('reconcile'):
-            move_lines = self.env['account.move.line'].search([('account_id', 'in', self.ids)], limit=1)
-            if len(move_lines):
-                raise UserError(_('You cannot change the value of the reconciliation on this account as it already has some moves'))
+        if 'reconcile' in vals:
```

Comment:
```python
if not vals.get('reconcile'):
    self.filtered(lambda x: x.reconcile)._toggle_reconcile_to_false()
else:
    self.filtered(lambda x: not x.reconcile)._toggle_reconcile_to_true()
```

Reply:
if **reconcile** is not present in the values to write, no sql query is needed.

Reply:
that wasn't my point: you can't do self.reconcile as write is an @api.multi function. So you need to filter records on which you want to call the toggle functions

Reply:
done in https://github.com/odoo/odoo/pull/19204/commits/810f1baa560ea810a4cdae8ba092b2e20d32d74c